indexer-core/src/main/java/org/apache/maven/index/updater/DefaultIndexUpdater.java

```diff
@@ -190,9 +191,7 @@ private Date loadIndexDirectory( final IndexUpdateRequest updateRequest, final R
         {
             updateRequest.getIndexTempDir().mkdirs();
         }
-        File indexDir = File.createTempFile( remoteIndexFile, ".dir" , updateRequest.getIndexTempDir() );
-        indexDir.delete();
-        indexDir.mkdirs();
+        File indexDir = Files.createTempDirectory( updateRequest.getIndexTempDir().toPath(), remoteIndexFile ).toFile();
```

Comment:
updateRequest.getIndexTempDir() is nullable, above is nullcheck performed. Better make `File indexDir` above IF and then update it in IF (add ELSE as well).